### DISCUSSION TITLE
Trying to setup a macro that detects Keyboard Hotkey release.

### DISCUSSION THREAD

**Discussion by GuiThink** [WarmUpTill/SceneSwitcher#612] [Q&A] [answered]
Hello,

I've been trying to setup SceneSwitcher so when I keep pressing a Hotkey "G" it keeps showing a Scene "1" as long as the Hotkey "G" is pressed. On release of Hotkey "G" it should then switch to a Scene "2".
Is it possible to perform such thing ?

Thank you for your help :)

**Comment by WarmUpTill** [comment, marked as answer]
Hello,

with the current implementation of the hotkey condition that is not possible.
I slightly tweaked the implementation to allow this with this PR:
https://github.com/WarmUpTill/SceneSwitcher/pull/613

A build with this change should be available here in a few minutes:
https://github.com/WarmUpTill/SceneSwitcher/actions/runs/3607052462

Using this build you can do something like this to achieve the effect you are looking for:

![Scene1](https://user-images.githubusercontent.com/19472752/205424351-f745ab2c-1437-42ba-8fd0-b68c16865531.PNG)

![Scene2](https://user-images.githubusercontent.com/19472752/205424297-22aab177-8e63-4edc-bd13-7a9d4677d4c3.PNG)

Let me know if you have any questions or run into any issues! :)

**Reply by GuiThink** [reply]
Hello !

This is actually incredible and it works just as expected !!! You are awesome ! Thank you so much for adding this feature so quick :)

**Reply by GuiThink** [reply]
Actually I found an expected behavior :)

Here is some context : when multiple keys are pressed at the same time, it seems the program do not detect that the intended Hotkey that triggers the scene is still constantly pressed. My feeling is that other key presses interfere with the Hotkey that is permanently pressed.  

For example, if I want to switch scene while playing a game using W/S/A/D keys, which are the standard keys for movement in games like FPS, then it seems the continuous pressing of the hotkey that triggers the scene is not detected anymore while I move my character in the game.

If I take your screenshots, let's say that "Macro trigger hotkey 1" = 'G'.
If I press 'G' itself, then it works and the scene changes.
If I press 'G' itself while also pressing keys like 'W', then 'S', then 'W' again, then 'A.'..or whatever key, then the scene does not change anymore.

Do you think this can be solved or it's getting too complicated ?

**Reply by WarmUpTill** [reply]
I cannot seem to reproduce that issue.
So that might actually be a hardware limitation of your keyboard (See [Keyboard Ghosting](https://www.microsoft.com/applied-sciences/projects/anti-ghosting-demo)).

Or you start accidentally adding modifier keys like the Shift key, which might also lead to the situation that the hotkey condition is no longer true, as you are now pressing "Shift + G" instead of just "G".
For the latter case you could simply add multiple hotkeys combined with the "Or" logic.

![Shift](https://user-images.githubusercontent.com/19472752/205478688-18724759-0b0b-4c1a-ac5d-d78f5684c125.PNG)

**Reply by GuiThink** [reply]
Hello !

The problem was the fact I was using shift indeed ! Good catch !
So it works just fine now with your last example :)

 Thank you so much for your help ! This plugin is amazing !
